src/deblock.rs

```diff
@@ -196,58 +226,133 @@ fn filter_wide14(p6: i32, p5: i32, p4: i32, p3: i32, p2: i32, p1: i32, p0: i32,
                                                p0   + q0   + q1   + q2   + q3   + q4*2 + q5*2 + q6*7 + (1<<3) >> 4]
 }   
 
-fn mask4(blimit: i32, limit: i32, p1: i32, p0: i32, q0: i32, q1: i32) -> bool {
-    (p1 - p0).abs() <= limit &&
-        (q1 - q0).abs() <= limit &&
-        (p0 - q0).abs() * 2 + (p1 - q1).abs() / 2 <= blimit
-}
-
 fn stride_copy(dst: &mut [u16], src: &[i32], pitch: usize) {
     for (dst, src) in dst.iter_mut().step_by(pitch).take(src.len()).zip(src) { *dst = *src as u16 };
 }
 
+fn stride_sse(a: &[u16], b: &[i32], pitch: usize) -> i64 {
+    let mut acc: i32 = 0;
+    for (a, b) in a.iter().step_by(pitch).take(b.len()).zip(b) {
```

Comment:
the `take()` could be omitted if it doesn't act as compiler hint.

Reply:
zip() doesn't care if the slices are different sizes?